apps/aecore/src/aec_chain_state.erl

```diff
@@ -146,21 +146,21 @@ insert_block(#{ key_block := KeyBlock, micro_blocks := MicroBlocks, dir := forwa
     end;
 insert_block(#{ key_block := KeyBlock, micro_blocks := MicroBlocks, dir := backward }) ->
     %% First insert micro_blocks
-    case lists:foldl(fun(MB, ok) -> insert_block(MB);
-                        (MB, {pof, _}) -> insert_block(MB);
+    case lists:foldl(fun(MB, ok) -> do_insert_block(MB, sync);
+                        (MB, {pof, _}) -> do_insert_block(MB, sync);
                         (_MB, Err = {error, _}) -> Err
                      end, ok, MicroBlocks) of
         ok ->
-            insert_block(KeyBlock, sync);
+            do_insert_block(KeyBlock, sync);
         {pof,_PoF} ->
-            insert_block(KeyBlock, sync);
+            do_insert_block(KeyBlock, sync);
         Err = {error, _} ->
             Err
     end;
 insert_block(Block) ->
```

Comment:
This keeps the check for:
* Key blocks received by gossip
* Micro blocks received by gossip
* Locally mined key block
* Locally signed micro block
* Block posted by user HTTP API (this is debatable though out of scope for this PR)

Ok.